buildlib/fedora.Dockerfile

```diff
@@ -0,0 +1,21 @@
+# docker build -t ucfconsort.azurecr.io/ucx/fedora:30 -f buildlib/fedora.Dockerfile buildlib/
+FROM fedora:30
+
+RUN dnf install -y \
+    autoconf \
+    automake \
+    doxygen \
+    file \
+    gcc-c++ \
+    git \
+    glibc-devel \
+    libtool \
+    make \
+    maven \
+    numactl-devel \
+    rdma-core-devel \
+    rpm-build \
+    clang \
+    cppcheck \
+    bear \
```

Comment:
Keep sorted lists sorted

Reply:
It looks like a lot of stuff is not building, I think more software needs to be setup:

```
configure: =========================================================
configure: UCX build configuration:
configure:       Multi-thread:   Disabled
configure:          MPI tests:   Disabled
configure:      Devel headers:   
configure:        UCT modules:   < ib rdmacm cma >
configure:       CUDA modules:   < >
configure:       ROCM modules:   < >
configure:         IB modules:   < >
configure:        UCM modules:   < >
configure:       Perf modules:   < >
configure: =========================================================
```

And you'll need to install a newer rdma-core..

Reply:
We should be able to install virtually everything, including CUDA, ROCM, etc. @yosefe can we drop in ugni headers just for compilation ?

Reply:
Currently, we build in default RHEL env using ~only required packages. Other configurations will be added later.

Reply:
You should focus this step on the pipeline to obtain full coverage and static results. Distro focused tests can do other things

Reply:
This entire PR is to have coverage of what `csmock` does. Next steps will provide coverage and add more distros.

Reply:
Personally I'd start with full coverage and then do limited coverage testing in the distro builds. 

IMHO no particularly good reason to do a full static sweep on anything other than full builds.

Again, I would largely ignore things like csmock and distros at this point. Focus on getting a CI that makes sense for UCX, then consider what more things might possibly be needed.

UCX needs a full compile CI, that is obviously critical..

Reply:
UCX has existing CI system, and the goal of this PR to extend it for RHEL checks. Full compile CI as a replacement of existing one as well as stricter compilation flags are the next steps.